srv/components/provisioner/config/refresh_machine_id.sls

```diff
@@ -0,0 +1,46 @@
+#
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+
+{% set old_machine_id = grains['machine_id'] %}
+
+Delete old machine_id:
+  file.absent:
+    - name: /etc/machine-id
+
+Refresh machine_id on {{ grains['id'] }}:
+  cmd.run:
+    - name: systemd-machine-id-setup
+
+Sync data:
+  module.run:
+    - saltutil.refresh_grains: []
```

Comment:
I have strong concerns that refreshed grains would be visible in later stages

Reply:
splitted sls into 2 files and added issue in our docs